### DISCUSSION TITLE
API与主机监控需求： 是否支持开源日志采集工具以及RESTful API接入?

### DISCUSSION THREAD

**Discussion by drunklab** [apache/iotdb#3955] [Q&A] [answered]
IoTDB 老师，您们好：）

一、咨询问题：
目前是否支持开源日志采集工具（filebeat,Flume、Logstash , Fluentd）以及RESTful API接入?
这两天大致看过IoTDB文档，了解到【应用编程接口】提供了java,go,Python接入以及SQL(CLI) ，但未查询到轻量级接入方式如RESTful API。

二、项目需求：
1、API接口质量监控与告警：
如API成功总数、成功率、失败总数，失败率、API总交易量、平均响应时间、QPS、按调用系统统计交易量 等等，并根据各指标预警。

2、主机监控与告警：
资源状态利用率指标监控、并根据阈值告警。

三、系统现状：
目前系统已在日志中输出了api指标信息：cost:耗时,seq:流水号,gnb:全局流水,apicode编号,reqappcd:请求系统,destappcd:服务接收系统,retcd:返回码,tmstamps:时间戳。
基于性能考虑计划通过异步日志采集工具（filebeat,Flume、Logstash , Fluentd）发送到IoTDB。

根据上述情况，如何通过云原生日志采集工具或轻量级方式快速集成IoTDB?   恳请提供一下这方面的帮助，非常感激！！

**Comment by HTHou** [comment, marked as answer]
您好，RESTful api 目前还没有，不过在我们的开发计划之内，感谢支持！
期望可以关注我们之后的新版本发布！

**Reply by HTHou** [reply]
https://iotdb.apache.org/zh/UserGuide/Master/API/RestService.html

Rest 在0.13.0版本里发布出来了

**Comment by drunklab** [comment]
您好，我们计划什么阶段发布 RESTful api？ 非常期待。
